src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -56,6 +57,15 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
   }
 
+  def fromNonEmptyString[T: ClassTag](fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
+    lazy val typeName = implicitly[ClassTag[T]]
```

Comment:
I really like the idea of using the `ClassTag`. Kudos.
My preference would be to use a `val` or a `def` instead of a `lazy val` given its overhead. The conventional wisdom I subscribe to is that unless the value is very expensive to compute or that you need `lazy val` for correctness, don't use `lazy val`.